src/Carousel.js

```diff
@@ -280,7 +280,7 @@ const Carousel = React.createClass({
       direction = direction || this.getDirection(previousActiveIndex, index);
 
       if (this.props.onSelect) {
-        this.props.onSelect(index, direction);
+        this.props.onSelect(index, e, direction);
```

Comment:
`event`? I thought about assigning `direction` there, but React actually pools synthetic events, and the pooling code doesn't clear unknown properties, so it'd pollute the object pool in a weird way, which seemed bad.

We could make the third callback argument an object, but I don't think there's much reason to do that – the meaning would be specific to the individual component, and I don't see any use cases other than `direction` for `<Carousel>`.

Alternatively, we could just drop `direction` in the callback altogether.